addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js

```diff
@@ -90,6 +90,150 @@ class EnvironmentScService extends Service {
     return this.augmentWithConnectionInfo(requestContext, envs);
   }
 
+  async pollAndSyncWsStatus(requestContext) {
+    const [indexesService, awsAccountsService] = await this.service(['indexesService', 'awsAccountsService']);
+    let envs = await this._scanner()
+      .limit(10000)
+      .scan();
+    envs = _.filter(
+      envs,
+      env => _.includes(['COMPLETED', 'STARTING', 'STOPPING', 'TERMINATING'], env.status) && env.inWorkflow !== 'true',
```

Comment:
nit: Is there a PENDING status that should be added as well. I think a comment here on which statuses are being considered and which are being discarded and why would be useful.

Reply:
PENDING is when environment provisioning in progress or pending, it should always happen during a workflow and the instance might not exist yet. So here it should not include PENDING. I'll add comment here explaining why certain statuses are included and others are excluded.